File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        stack<pair<int, int>> st;
        vector<int> result(temperatures.size(), 0);
        for(int i=0; i<temperatures.size(); i++){
            int t = temperatures[i];

            while(!st.empty() && t > st.top().first){
                auto top = st.top();
                result[top.second] = i - top.second;
                st.pop();
            }
            st.push({t, i});
        }
        return result;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] result = new int[2];
        for(int i=0, j=numbers.length-1; i<numbers.length;){
            if(numbers[i] + numbers[j] > target){
                j--;
                continue;
            }
            else if(numbers[i] + numbers[j] < target){
                i++;
                continue;
            }
            result[0] = i+1;
            result[1] = j+1;
            break;
        }

        return result;
    }
}
